### ISSUE TITLE
Use Camunda's defaults for Job Executor Thread Pool Size

### ISSUE THREAD

**Issue by falko** [camunda/camunda-bpm-spring-boot-starter#276]
According to the [Spring ThreadPoolExecutor Documentation](https://docs.spring.io/spring/docs/3.0.x/javadoc-api/org/springframework/scheduling/concurrent/ThreadPoolTaskExecutor.html#setCorePoolSize(int)) the default core pool size is 1 and the default maximum pool size is Integer.MAX_VALUE. Especially, the high maximum pools size is dangerous, because it could cause a too eager job acquisition on some cluster nodes while others do not get any jobs. Such an uneven job distribution leads to a poor overall resource utilization, longer process durations, and potentially duplicate side effects when jobs reach their lock timeout and get reacquired by other nodes of the cluster.

Long story short, the Spring Boot Starter should not work with the defaults of Spring's ThreadPoolTaskExecutor and instead use [Camunda's Default Job Execution Properties](https://docs.camunda.org/manual/latest/reference/deployment-descriptors/tags/job-executor/#job-executor-configuration-properties), i.e. corePoolSize=3 and maxPoolSize=10.

**Comment by jangalinski** [comment]
Is it possible to get the defaults from camunda config to avoid redundancy?

**Comment by jangalinski** [comment]
see #278

**Comment by falko** [comment]
@jangalinski Thx, for fixing this so quickly! Regarding your question, we are currently debating a default configuration class with all the sensible defaults: https://app.camunda.com/jira/browse/CAM-7606

**Comment by jangalinski** [comment]
That would indeed be a good idea ... in the starter, I create a new Configuration instance, just to access its defaults.
